primo/data_parser/metric_data.py

```diff
@@ -0,0 +1,602 @@
+                f"which is not a valid python variable name!"
+            )
+
+        self._name = val
+
+    @property
+    def value(self):
+        """Getter for attribute `value`"""
+        return self._value
+
+    @value.setter
+    def value(self, val):
+        """Setter for attribute `value`"""
+        if self.min_value <= val and val <= self.max_value:
+            self._value = val
+            return
+
+        raise ValueError(
+            f"Attempted to assign {val} for metric {self.name}, which "
+            f"lies outside the admissible range "
+            f"[{self.min_value}, {self.max_value}]."
+        )
+
+    @property
+    def weight(self):
+        """Getter for attribute `weight`"""
+        # NOTE: For primary metrics, weight = value. But for submetrics,
+        # weight = (value / 100) * parent_metric.weight
+        return self._value
+
+    @property
+    def score_col_name(self):
+        """Getter for attribute `score_col_name`"""
+        if self._score_col_name is not None or self.data_col_name is None:
+            return self._score_col_name
+
+        self._score_col_name = self.data_col_name + f" Score [0-{self.weight}]"
+        return self._score_col_name
+
+
+class SubMetric(Metric):
+    """
+    Dataclass for creating submetric objects
+    """
+
+    def __init__(
+        self,
+        name,
+        parent_metric,
+        value,
+        min_value=0,
+        max_value=100,
+        full_name=None,
+    ) -> None:
+        """
+        Returns a `SubMetric` object
+
+        Parameters
+        ----------
+        name : str
+            Short name of the metric.
+            Must be a valid python variable name
+
+        parent_metric : Metric
+            Parent metric object
+
+        value : float
+            Weight associated with the metric
+
+        min_value : float, default = 0
+            Lower bound on metric value
+
+        max_value : float, default = 100
+            Upper bound on metric value
+
+        full_name : str, default = None
+            Elaborate name of the metric
+
+        Raises
+        ------
+        ValueError
+            If an invalid python variable name is provided,
+            If `value` lies outside the bounds,
+            If an attempt is made to overwrite `name`
+        """
+
+        super().__init__(name, value, min_value, max_value, full_name)
+        self.is_submetric = True
+        self.parent_metric = parent_metric
+
+    def __str__(self) -> str:
+        """Format for printing objects"""
+        return (
+            f"Submetric name: {self.full_name}, Submetric value: {self.value} \n"
+            f"    Admissible range: [{self.min_value}, {self.max_value}] \n"
+            f"    Is a submetric of {self.parent_metric.full_name}"
+        )
+
+    @property
+    def weight(self):
+        """Getter for attribute `weight`"""
+        return (self.value / 100) * self.parent_metric.weight
```

Comment:
The weight for a submetric could refer to either:
- weight relative to other submetrics in the same parent metric class, or
- absolute weight in total metric calculations

Could you update the docstring to clarify what exactly this property helps identify? Also might be useful to help create another property method for the other weight